payroll/models/hr_payslip.py

```diff
@@ -281,25 +284,11 @@ def compute_sheet(self):
         for payslip in self:
             # delete old payslip lines
             payslip.line_ids.unlink()
-            # set the list of contract for which the rules have to be applied
-            # if we don't give the contract, then the rules to apply should be
-            # for all current contracts of the employee
-            contract_ids = (
-                payslip.contract_id.ids
-                or payslip.employee_id._get_contracts(
-                    date_from=payslip.date_from, date_to=payslip.date_to
-                ).ids
-            )
             # write payslip lines
             number = payslip.number or self.env["ir.sequence"].next_by_code(
                 "salary.slip"
             )
-            lines = [
-                (0, 0, line)
-                for line in list(
-                    self._get_payslip_lines(contract_ids, payslip.id).values()
-                )
-            ]
+            lines = [(0, 0, line) for line in list(payslip.get_lines_dict().values())]
```

Comment:
You are right, but _get_pasylip_lines() has been around for a while and other modules most likely have inherited it. So, I don't want to break them. However, I've included a warning whenever they call it so that can update their code.